react/headless/src/utils/directive.ts

```diff
@@ -49,3 +51,35 @@ export function useDirectives<T>(directives: Directive<T>[], args?: T): RefCallb
 	const mergedDirectives = useMemo(() => mergeDirectives(...directives), directives);
 	return useDirective(mergedDirectives, args as any);
 }
+
+const attributesMap = new Map([
+	['tabindex', 'tabIndex'],
+	['for', 'htmlFor'],
+]);
+
+/**
+ * Returns an object with the key/value attributes for JSX, derived from a list of directives.
+ *
+ * @param directives - List of directives to generate attributes from. Each parameter can be the directive or an array with the directive and its parameter
+ * @returns JSON object with key/value pairs to be applied on a JSX node.
+ */
+export function ssrAttributes<T extends any[]>(...directives: {[K in keyof T]: AttributeDirectiveAndParam<T[K]> | AttributeDirective<void>}) {
```

Comment:
In the core, the `ssrAttributes` function is just a shortcut to the `directiveAttributes` function depending on the environment. It should be the same in react.